src/Connection/Http1Connection.php

```diff
@@ -87,7 +95,11 @@ public function request(Request $request, ?CancellationToken $cancellation = nul
             try {
                 yield RequestWriter::writeRequest($this->socket, $request, $protocolVersion);
 
-                return yield from $this->doRead($request, $cancellation, $readingCancellation, $completionDeferred);
+                $response = yield from $this->doRead($request, $cancellation, $readingCancellation, $completionDeferred);
+
+                $this->busy = false;
```

Comment:
This is too early. You need to subscribe to the completion promise in `Response` instead, because at this point we have only read the header, not the body.

Reply:
This was intentional, since another request can be made while a prior response body is being streamed… but that could cause another request to be severely delayed, so I'll use the completion promise instead.

Reply:
As far as I know no browser supports pipelining, so we probably shouldn't either.

Reply:
We do in `http-server` (because we should to comply with the RFC I think…), but in the client, I agree.

Reply:
Yes, for `http-server` we need to support it.